Network module

muhamadazmy opened the pull request:
Network module

muhamadazmy committed:
Add specs for networking

muhamadazmy committed:
Add specs for networking

muhamadazmy committed:
Add specs for networking

muhamadazmy committed:
Add specs for networking

muhamadazmy commented:
After a small talk with @delandtj  i think the networking responsibilities should be different as follows:
- The network module/daemon is responsible for the networking of the host, no more.
- CNI is called by the runtime, we don't need to interfere except may be create a new plugin when needed.
- To make this consistent with the way we run VMs, the same plugins still can be invoked by the vm module to setup same network infrastructure. 
- This means of curse, we shouldn't use libvirt anymore for VMs, but @delandtj thinks that we really should do that anyway. I definitely agree with him on that.

zaibon committed:
add summary of brainstoming session about networking

zaibon committed:
add summary of brainstoming session about networking

zaibon committed:
add summary of brainstoming session about networking

zaibon committed:
add summary of brainstoming session about networking

delandtj committed:
requirements

delandtj committed:
requirements

zaibon committed:
add lifecyle schema + links to container runtime documentations

delandtj committed:
progress

delandtj committed:
progress

delandtj committed:
continuing from home

delandtj committed:
continuing from home

delandtj committed:
w

delandtj committed:
w

delandtj committed:
pffff

delandtj committed:
pffff

zaibon committed:
Merge branch 'network_addon' into networking

zaibon committed:
move Requirements.md into specs/network folder

zaibon committed:
move Requirements.md into specs/network folder

muhamadazmy committed:
Merge branch 'master' into networking

delandtj committed:
fikzd

delandtj committed:
fikzd

delandtj committed:
questions

delandtj committed:
questions

delandtj committed:
test svg

delandtj committed:
test svg

delandtj committed:
test svg

delandtj committed:
test svg

zaibon committed:
organize flist module

- split flist package into lib and daemon
- move flist daemon code to flistd directory
- add client.go that allow to get a Flister easily by other libs

zaibon committed:
add containerd client example

- the example starts a caddy container using an flist and containerd

Commit:
WIP: building the container module

delandtj committed:
netresource

delandtj committed:
netresource

muhamadazmy committed:
Update readme

delandtj committed:
questions, some more descriptions

delandtj committed:
questions, some more descriptions

muhamadazmy committed:
WIP: container modules specs

delandtj committed:
some udates/corrections

delandtj committed:
some udates/corrections

zaibon commented on specs/network/cases.md:
```diff
@@ -0,0 +1,128 @@
+### QUESTIONS:
+
+  - Storage of network objects.
+  It has been said that we will use the 'blockchain' to store 'things'. A network/network resource could very well be such a 'thing', but at the moment there is no clear view on how to do that and how big these 'things' can be. Also, that 'thing' needs to be encrypted to make sure we don't enlarge a possible attack surface towards a tenant network.
```
As far as I understand the "blockchain" used to store data won't be a "real" blockchain, but just a bunch of BCDB with some RAFT on top for consensus.
So storage wise we should not have a problem cause BCDB can store big amount of data.
Now for notification, seems we need to think how we could create something that doesn't require all the node too keep doing some pooling on the BCDB, that would requires way to much bandwidth.
Maybe we could open a FR in BCDB to have some kind of publish/subcribe mechanism so any change can be pushed the the required nodes.
Needs further thinking on this.

delandtj committed:
Let's start with that

delandtj committed:
Let's start with that

delandtj committed:
formatting

delandtj committed:
formatting

delandtj committed:
formatting

delandtj committed:
formatting

delandtj committed:
Update cases.md

delandtj committed:
Update cases.md

muhamadazmy committed:
WIP: add namespaces

delandtj committed:
updated readme.md

delandtj committed:
updated readme.md

delandtj committed:
oops forgot to save

delandtj committed:
oops forgot to save

delandtj committed:
same

delandtj committed:
same

delandtj committed:
Merge branch 'networking' of github.com:threefoldtech/zosv2 into networking

muhamadazmy committed:
Implement delete container

muhamadazmy committed:
Handle mounts

muhamadazmy committed:
add readme

delandtj committed:
Tenent Network application

delandtj committed:
Tenent Network application

delandtj committed:
typo

delandtj committed:
typo

delandtj committed:
png

delandtj committed:
png

delandtj committed:
integrate pic

delandtj committed:
integrate pic

delandtj committed:
rename storage -> TNoDB
NRR still needs naming love

delandtj committed:
rename storage -> TNoDB
NRR still needs naming love

delandtj committed:
airport

delandtj committed:
airport

delandtj committed:
Merge branch 'networking' of github.com:threefoldtech/zosv2 into networking

delandtj committed:
some updates

delandtj committed:
some updates

muhamadazmy committed:
Apply review comments

Commit:
Update modules/container/container.go

Co-Authored-By: Christophe de Carvalho <zaibon@users.noreply.github.com>

delandtj committed:
tnodb

delandtj committed:
tnodb

muhamadazmy committed:
Implement inspect

Apply some review comments

delandtj committed:
some adds

delandtj committed:
some adds

maxux committed:
Update container README with new arguments

maxux committed:
Update examples

maxux committed:
Update example files name

zaibon committed:
container module: decouple flist and container modules

container module has no notion of flist, only paths
this allow to remove the dependency of flist module on the container
module

zaibon committed:
container module: add support for coreX in the container

- change entrypoint to coreX binary
- mount flist as /sandbox and make coreX chroot into /sandbox at start

zaibon committed:
add contctl debugging tool

zaibon committed:
WIP network

delandtj committed:
Merge branch 'networking' of github.com:threefoldtech/zosv2 into networking

zaibon committed:
network module: implement network bootstrap

- the network module does loop over all the connected interfaces
to try to get an IP.

IPv6 is configured via SLAAC
IPv4 is configured via DHCP

zaibon committed:
network module: add networker interface

- also add types used by the network module itself

zaibon committed:
Implement zinit client

zaibon committed:
use yaml to parse zinit outputs

zaibon committed:
Create notes_20191306.md

zaibon committed:
Update notes_20191306.md

delandtj committed:
Merge branch 'networking' of github.com:threefoldtech/zosv2 into networking

delandtj committed:
setuptests

delandtj committed:
ran 255 for testing

zaibon commented:
this branch got out of controle closing in favor of: https://github.com/threefoldtech/zosv2/pull/49
